dev/DynamicDependency/M.AM.DD.PackageDependency.cpp

```diff
@@ -0,0 +1,89 @@
+        wil::unique_process_heap_string packageDependencyId;
+        THROW_IF_FAILED(MddTryCreatePackageDependency(nullptr, packageFamilyName.c_str(), mddMinVersion, mddArchitectures, mddLifetimeKind, nullptr, mddOptions, wil::out_param(packageDependencyId)));
+        return GetFromId(packageDependencyId.get());
+    }
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::CreateForSystem(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion, Microsoft::ApplicationModel::DynamicDependency::CreatePackageDependencyOptions const& options)
+    {
+        const auto mddMinVersion = ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion);
+        const auto mddArchitectures = ::Microsoft::ApplicationModel::DynamicDependency::ToArchitectures(options.Architectures());
+        const auto mddLifetimeKind = ::Microsoft::ApplicationModel::DynamicDependency::ToLifetimeKind(options.LifetimeArtifactKind());
+        auto mddOptions = ::Microsoft::ApplicationModel::DynamicDependency::ToCreateOptions(options, MddCreatePackageDependencyOptions::ScopeIsSystem);
+        wil::unique_process_heap_string packageDependencyId;
+        THROW_IF_FAILED(MddTryCreatePackageDependency(nullptr, packageFamilyName.c_str(), mddMinVersion, mddArchitectures, mddLifetimeKind, nullptr, mddOptions, wil::out_param(packageDependencyId)));
+        return GetFromId(packageDependencyId.get());
+    }
+    hstring PackageDependency::Id()
+    {
+        return m_id;
+    }
+    void PackageDependency::Delete()
+    {
+        MddDeletePackageDependency(m_id.c_str());
+    }
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependencyContext PackageDependency::Add()
+    {
+        const auto rank = MDD_PACKAGE_DEPENDENCY_RANK_DEFAULT;
+        const auto mddOptions = MddAddPackageDependencyOptions::None;
+        wil::unique_package_dependency_context packageDependencyContext;
+        THROW_IF_FAILED(MddAddPackageDependency(m_id.c_str(), rank, mddOptions, &packageDependencyContext, nullptr));
+        auto context = winrt::make<implementation::PackageDependencyContext>(packageDependencyContext.get());
+        packageDependencyContext.release();
+        return context;
+    }
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependencyContext PackageDependency::Add(Microsoft::ApplicationModel::DynamicDependency::AddPackageDependencyOptions const& options)
+    {
+        const auto rank = options.Rank();
+        const auto mddOptions = ::Microsoft::ApplicationModel::DynamicDependency::ToAddOptions(options);
+        wil::unique_package_dependency_context packageDependencyContext;
+        THROW_IF_FAILED(MddAddPackageDependency(m_id.c_str(), rank, mddOptions, &packageDependencyContext, nullptr));
+        auto context = winrt::make<implementation::PackageDependencyContext>(packageDependencyContext.get());
+        packageDependencyContext.release();
```

Comment:
Use `move` here:

```
auto context = winrt::make<implementation::PackageDependencyContext>(std::move(packageDependencyContext));
```

This pattern is a signal you should be moving during ownership transfer:

```c++
auto x = thing_that_creates(context-thing.get());
context-thing.release();
```